modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/internal/order/CommerceOrderHttpHelperImpl.java

```diff
@@ -582,9 +582,15 @@ private CommerceOrder _checkGuestOrder(
 		if (userCommerceOrder == null) {
 			httpSession.removeAttribute(cookieName);
 
-			return _commerceOrderLocalService.updateAccount(
+			commerceOrder = _commerceOrderLocalService.updateAccount(
 				commerceOrder.getCommerceOrderId(), user.getUserId(),
 				accountEntry.getAccountEntryId());
+			commerceOrder = _commerceOrderLocalService.updateShippingAddress(
+				commerceOrder.getCommerceOrderId(), 0);
+			commerceOrder = _commerceOrderLocalService.updateBillingAddress(
+				commerceOrder.getCommerceOrderId(), 0);
+
```

Comment:
You are re-indexing twice unnecessarily if you do this. Following is better if we do not need to reindex:

```
commerceOrder.setBillingAddressId(0);
commerceOrder.setShippingAddressId(0);

 _commerceOrderLocalService.updateAccount(
				commerceOrder.getCommerceOrderId(), user.getUserId(),
				accountEntry.getAccountEntryId());

```

Reply:
set*Address methods only update the object, the db entity is not updated. So when updateAccount is called, the order is fetched with the original addresses' ids. We need to find a way to update the address id values in the db as well

Reply:
@brianikim Could we make a resetAddresses method like [resetTermsAndConditions](https://github.com/liferay/liferay-portal/blob/6c18f8775a043f2ad6f56daddda998a2542acfef/modules/apps/commerce/commerce-service/src/main/java/com/liferay/commerce/service/impl/CommerceOrderLocalServiceImpl.java#L1056)?
It would set the addresses' ids to 0 and update in the db